functional-tests/src/main/java/com/redhat/osio/util/RestClient.java

```diff
@@ -0,0 +1,129 @@
+/*
+ * Copyright (c) 2016-2018 Red Hat, Inc.
+ * This program and the accompanying materials are made
+ * available under the terms of the Eclipse Public License 2.0
+ * which is available at https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *   Red Hat, Inc. - initial API and implementation
+ */
+package com.redhat.osio.util;
+
+import com.google.gson.JsonObject;
+import java.io.IOException;
+import java.util.Map.Entry;
+import java.util.Set;
+import java.util.concurrent.TimeUnit;
+import okhttp3.MediaType;
+import okhttp3.OkHttpClient;
+import okhttp3.Request;
+import okhttp3.Request.Builder;
+import okhttp3.RequestBody;
+import okhttp3.Response;
+import org.apache.http.entity.ContentType;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class RestClient {
+
+  private static final Logger LOG = LoggerFactory.getLogger(RestClient.class);
+
+  private OkHttpClient client;
+  private String serverURL;
+
+  public RestClient(String serverURL) {
+    this.serverURL = serverURL;
+    this.client = new OkHttpClient.Builder().build();
+  }
+
+  public RestClient(String serverURL, long timeout, TimeUnit timeUnit) {
+    this.serverURL = serverURL;
+    this.client = new OkHttpClient.Builder().readTimeout(timeout, timeUnit).build();
+  }
+
+  public void close() {
+    try {
+      client.dispatcher().executorService().shutdown();
+      client.connectionPool().evictAll();
+      client.cache().close();
+    } catch (Exception e) {
+      LOG.info("OkHttpClient threw exception when closing connection:" + e.getMessage(), e);
+    }
+  }
+
+  public Response sendRequest(HttpMethods method) throws IOException {
+    return sendRequest(null, method);
+  }
+
+  public Response sendRequest(String relativePath, HttpMethods method) throws IOException {
+    return sendRequest(relativePath, method, null);
+  }
+
+  public Response sendRequest(String relativePath, HttpMethods method, JsonObject body)
+      throws IOException {
+    return sendRequest(relativePath, method, body, null, null);
```

Comment:
Wouldn't it be better to use this line in methods above too? Instead of sending
(null, method) and (relativePath, method, null) ? There is no logic or setting in these methods so I can't see any reason to go through all of them.

Reply:
It clearly shows what is set to null in the specific call. Other values can, but don't have to be null in them. In my opinion it's cleaner like this, but I can change this if requested.

Reply:
@rhopp what do you think? It seems for me like slowdown - call method with method set which will call next method to set body to null, then next method to set next part to null... Wouldn't it be better to call just (null, method, null, null, null) to avoid this chain of methods (which do only set to null)?